From 12a52402214d3ab0693b790e9e94bf5c96c16164 Mon Sep 17 00:00:00 2001
From: Harshitha Prem <quic_hprem@quicinc.com>
Date: Tue, 20 Dec 2022 16:28:15 +0530
Subject: [PATCH] ath12k: remove paging_dyn section from
 paging.bin

When firmware dumps are collected, paging.bin parsed
in rddm dumps is approximately 18MB but the expectation
from firmware is only 6MB.

This difference in size is seen because of paging_dyn.bin
information being appended to paging.bin.Hence, add changes
to handle the same

Signed-off-by: Harshitha Prem <quic_hprem@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.h     |  3 ---
 drivers/net/wireless/ath/ath12k/coredump.c | 19 -------------------
 drivers/net/wireless/ath/ath12k/qmi.c      |  4 ----
 3 files changed, 26 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/core.h b/drivers/net/wireless/ath/ath12k/core.h
index 5bd5584..931dcd1 100644
--- a/drivers/net/wireless/ath/ath12k/core.h
+++ b/drivers/net/wireless/ath/ath12k/core.h
@@ -1313,9 +1313,6 @@ void ath12k_core_halt(struct ath12k *ar);
 int ath12k_core_resume(struct ath12k_base *ab);
 int ath12k_core_suspend(struct ath12k_base *ab);
 int ath12k_config_qdss(struct ath12k_base *ab);
-int ath12k_coredump_mhi_update_bhie_table(struct ath12k_base *ab,
-					  void __iomem *va,
-					  phys_addr_t pa, size_t size);
 
 void ath12k_core_put_hw_group(struct ath12k_base *ab);
 
diff --git a/drivers/net/wireless/ath/ath12k/coredump.c b/drivers/net/wireless/ath/ath12k/coredump.c
index 56e7122..17e1e99 100644
--- a/drivers/net/wireless/ath/ath12k/coredump.c
+++ b/drivers/net/wireless/ath/ath12k/coredump.c
@@ -375,25 +375,6 @@ out:
 	vfree(dump);
 }
 
-int ath12k_coredump_mhi_update_bhie_table(struct ath12k_base *ab,
-					  void __iomem *va,
-					  phys_addr_t pa, size_t size)
-{
-	struct ath12k_pci *ar_pci = (struct ath12k_pci *)ab->drv_priv;
-	struct mhi_controller *mhi_ctrl = ar_pci->mhi_ctrl;
-	int ret;
-
-	/* Attach Pageable region to MHI buffer so that it is
-	 * included as part of pageable region in dumps
-	 */
-	ret = mhi_update_bhie_table_for_dyn_paging(mhi_ctrl, va, pa, size);
-	if (ret)
-	        ath12k_dbg(ab, ATH12K_DBG_QMI,
-			   "failed to add Dynamic Paging region to MHI Buffer table %d\n", ret);
-
-	return ret;
-}
-
 void ath12k_coredump_m3_dump(struct ath12k_base *ab,
 			     struct ath12k_qmi_m3_dump_upload_req_data *event_data)
 {
diff --git a/drivers/net/wireless/ath/ath12k/qmi.c b/drivers/net/wireless/ath/ath12k/qmi.c
index 962d88b..3f725a7 100644
--- a/drivers/net/wireless/ath/ath12k/qmi.c
+++ b/drivers/net/wireless/ath/ath12k/qmi.c
@@ -3511,10 +3511,6 @@ skip_mlo_mem_init:
 					ab->qmi.target_mem[i].size;
                         ab->qmi.target_mem[idx].type =
 					ab->qmi.target_mem[i].type;
-			ret = ath12k_coredump_mhi_update_bhie_table(ab,
-					ab->qmi.target_mem[idx].v.ioaddr,
-					ab->qmi.target_mem[idx].paddr,
-					ab->qmi.target_mem[idx].size);
 			sz += ab->qmi.target_mem[i].size;
 			if (ret < 0)
 				ath12k_warn(ab,
-- 
2.17.1

